generated/nidaqmx/_lib_time.py

```diff
@@ -58,33 +59,30 @@ def from_datetime(cls, dt: Union[std_datetime, ht_datetime]) -> AbsoluteTime:
 
         return AbsoluteTime(lsb=lsb, msb=timestamp_1904_epoch)
 
-    def to_datetime(self, tzinfo: Optional[timezone] = None) -> ht_datetime:
-        # First, calculate whole seconds by converting from the 1904 to 1970 epoch.
-        timestamp_1904_epoch = self.msb
-        was_positive = timestamp_1904_epoch > 0
-        timestamp_1970_epoch = int(timestamp_1904_epoch - AbsoluteTime._BIAS_FROM_1970_EPOCH)
-
-        # Our bias is negative, so our sign should only change if we were previously positive.
-        is_positive = timestamp_1970_epoch > 0
-        if is_positive != was_positive and not was_positive:
-            raise OverflowError(f"Can't represent {str(self)} in datetime (1970 epoch)")
+    def convert_to_desired_timezone(self, expected_time_utc, tzinfo):
+        current_time_utc = ht_datetime.now(timezone.utc)
+        desired_timezone_offset = current_time_utc.astimezone(tz=tzinfo).utcoffset()
+        desired_expected_time = expected_time_utc + desired_timezone_offset
+        new_datetime = ht_datetime(
+            desired_expected_time.year,
+            desired_expected_time.month,
+            desired_expected_time.day,
+            desired_expected_time.hour,
+            desired_expected_time.minute,
+            desired_expected_time.second,
+            desired_expected_time.microsecond,
+            desired_expected_time.femtosecond,
+            desired_expected_time.yoctosecond,
+            tzinfo = tzinfo)
+        return new_datetime
 
-        # Finally, convert the subseconds to micro, femto, and yoctoseconds.
+    def to_datetime(self, tzinfo: Optional[timezone] = None) -> ht_datetime:
         total_yoctoseconds = int(
             round(AbsoluteTime._YS_PER_S * self.lsb / AbsoluteTime._NUM_SUBSECONDS)
         )
-        microsecond, remainder_yoctoseconds = divmod(total_yoctoseconds, AbsoluteTime._YS_PER_US)
-        femtosecond, remainder_yoctoseconds = divmod(
-            remainder_yoctoseconds, AbsoluteTime._YS_PER_FS
-        )
-        yoctosecond = remainder_yoctoseconds
-
-        # Start with UTC
-        dt = ht_datetime.fromtimestamp(timestamp_1970_epoch, timezone.utc)
-        # Add in precision
-        dt = dt.replace(microsecond=microsecond, femtosecond=femtosecond, yoctosecond=yoctosecond)
-        # Then convert to requested timezone
-        return dt.astimezone(tz=tzinfo)
+        datetime_1904 = ht_datetime(1904, 1, 1, tzinfo=timezone.utc)
+        dt = datetime_1904 + ht_timedelta(seconds=self.msb) + ht_timedelta(yoctoseconds=total_yoctoseconds)
```

Comment:
I **think** we need to subtract the fractional seconds when msb is negative.

Reply:
After thinking about it for way too long, I don't think we should subtract the fractional seconds. 
Let's say the user enters 1903:12:31:0:0:0:....1ys. 
We would calculate this way: 1904 + (-1 day) + 1ys

To make it easier I translated this into simple math:
If we wanted to get to 5.01 from a value of 10, we would do it like this:
10 + (-5) + (0.01)
 
We will already use negative value for msb, and we truncate the value so we will get a whole number. Then we need to add back the user required fractional seconds.

Please let me know if there are any gaps in this logic